tests/unittest/_torch/visual_gen/test_teacache.py

```
@@ -69,3 +163,496 @@ def test_disabled_teacache_is_noop(self):
+            TeaCacheConfig(coefficients=[])
+
+    def test_empty_coefficients_2_rejected(self):
+        with pytest.raises(ValueError, match="coefficients_2"):
+            TeaCacheConfig(coefficients_2=[])
+
+    def test_negative_ret_steps_rejected(self):
+        with pytest.raises(ValueError, match="ret_steps must be non-negative"):
+            TeaCacheConfig(ret_steps=-1)
+
+    def test_cutoff_steps_exceeding_num_steps_rejected(self):
+        with pytest.raises(ValueError, match="cutoff_steps.*cannot exceed"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)